### PR TITLE
Draft 3.1.2 release blogpost

### PR DESCRIPTION
**Pull request by Kordyjan**

### CODE REVIEW

File: blog/_posts/2022-04-04-scala-3.1.2-released.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.2 released!
+---
+
+We are happy to announce the release of Scala 3.1.2. This new release brings a few significant improvements awaited by the community.
+
+## Configurable Scala output version
```
Comment (prolativ) [inline blog/_posts/2022-04-04-scala-3.1.2-released.md:10]: At this point the term `scala output version` might not yet be well understood as its origin is explained in the description of compiler flags. So how about renaming this section to something like `Forward compatible compilation`?

File: blog/_posts/2022-04-04-scala-3.1.2-released.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+First and foremost, after months of brainstorming and testing, we are bringing the possibility to configure the Scala output version to the stable release of the compiler.
```
[the remark this answers is not in the document]
Comment (prolativ) [reply blog/_posts/2022-04-04-scala-3.1.2-released.md:12]:
```suggestion
First and foremost, after months of brainstorming and testing, we are bringing the possibility to produce compilation outputs in a format used by a different (older) version of the compiler.
```

File: blog/_posts/2022-04-04-scala-3.1.2-released.md (hunk 1 above)
Comment (Kordyjan) [reply blog/_posts/2022-04-04-scala-3.1.2-released.md:10]: Now the first paragraph explains what `configurable scala output means`. Is it better?

File: blog/_posts/2022-04-04-scala-3.1.2-released.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.2 released!
+---
+
+We are happy to announce the release of Scala 3.1.2. This new release brings a few significant improvements awaited by the community.
+
+## Configurable Scala output version
+
+First and foremost, after months of brainstorming and testing, we are bringing the possibility to configure the Scala output version. This means that compiler can now generate TASTy files and classfiles that are readable for the older minor versions of Scala. Moreover, the minimal version of the compiler that is able to read your compiled code is controlled by single setting.
```
Comment (prolativ) [inline blog/_posts/2022-04-04-scala-3.1.2-released.md]:
```suggestion
First and foremost, after months of brainstorming and testing, we are bringing an experimental possibility to configure the Scala output version. This means that the compiler can now generate TASTy files and classfiles in a format used by an older minor version of Scala (which can currently be only 3.0.x) effectively specifying the minimal version of the compiler that is able to read your compiled code.
```

Comment (Kordyjan) [reply blog/_posts/2022-04-04-scala-3.1.2-released.md]: Applied manually.

File: _downloads/2022-04-08-3.1.2.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+title: Scala 3.1.1
```
Comment (SethTisue) [inline _downloads/2022-04-08-3.1.2.md:2]: 3.1.2

File: blog/_posts/2022-04-08-scala-3.1.2-released.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.2 released!
+---
+
+We are happy to announce the release of Scala 3.1.2. This new release brings a few significant improvements awaited by the community.
+
+## Configurable Scala output version
+
+First and foremost, after months of brainstorming and testing, we are bringing an experimental possibility to configure the Scala output version. This means that the compiler can now generate TASTy files and classfiles in a format used by an older minor version of Scala (which can currently be only 3.0.x) effectively specifying the minimal version of the compiler that is able to read your compiled code.
```
Comment (SethTisue) [inline blog/_posts/2022-04-08-scala-3.1.2-released.md:12]: s/minimal/minimum/

File: blog/_posts/2022-04-08-scala-3.1.2-released.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+If you are a library maintainer, you may have faced a dilemma, should you update the Scala version to 3.1 or stay on 3.0. On the one hand, updating would bring you all bug fixes and enhancements from the newest version of the compiler. On the other, all new releases of your library would be unusable for all the users that are, for some reason, stuck on Scala 3.0.
+
+With Scala 3.1.2, this tradeoff is no more. Now you can set the Scala output to previous language minor version. Using 3.1.2 to compile your library, you can set your output to Scala 3.0. Then your library can be used by code compiled with Scala 3.0, 3.1, or any future version. All produced TASTy files and classfiles will be compatible with Scala 3.0 and newer.
+
+While the flag itself is not marked as experimental, its role in the broader ecosystem of scala libraries is yet to be determined. We are eagerly awaiting feedback from the maintainers of the libraries. Hearing your opinions will allow us to choose the right strategy for supporting code compiled for previous scala versions so that we can find a balance between language development and burden on both compiler users and maintainers.
```
Comment (SethTisue) [inline blog/_posts/2022-04-08-scala-3.1.2-released.md:18]: s/scala/Scala/  (twice)

File: blog/_posts/2022-04-08-scala-3.1.2-released.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+### Current support
+
+You can set the Scala output version by using a new compiler flag named `-scala-output-version` with a minor release version as an argument. You may remember this flag as `-scala-target` or `-Xscala-release` in design discussions and unstable builds. Those names are not supported in the stable release.
+
+Sbt, starting from version 1.7.0-M1, has a setting `scalaOutputVersion` that not only sets the compiler flag mentioned above, but also fixes which version of the standard library will be specified as your dependency. For example, specifying
```
Comment (SethTisue) [inline blog/_posts/2022-04-08-scala-3.1.2-released.md:24]: s/fixes/determines/

File: blog/_posts/2022-04-08-scala-3.1.2-released.md (hunk 6 above)
Comment (julienrf) [reply blog/_posts/2022-04-08-scala-3.1.2-released.md:18]: When reading this, it is not clear to me what type of feedback you would like to get from the community.

File: _data/scala-releases.yml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 - category: current_version
   title: Current 3.1.x release
   version: 3.1.2
-  release_date: April 8, 2022
+  release_date: April 11 2022
```
Comment (Philippus) [inline _data/scala-releases.yml]:
```suggestion
  release_date: April 11, 2022
```

**Review by Philippus — CHANGES_REQUESTED** [review]

**Pull request merged**